api/compaction_manager.cc

```diff
@@ -74,14 +74,15 @@ void set_compaction_manager(http_context& ctx, routes& r) {
 
     cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<request> req) {
         return ctx.db.map_reduce0([&ctx](database& db) {
-            std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash> tasks;
+            return do_with(std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>(), [&ctx, &db](std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>& tasks) {
```

Comment:
@avikivity look at what do_with lines look like now that you outlawed auto& :-)

Reply:
How about:

     using pending_tasks_by_table_type = td::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>;

and now the code is shorter and clearer.

Reply:
@amnonh ping

Reply:
it's unrelated change, there's nothing wrong with using the full type, it's for the IDE and the IDE is happiest with full types.